### PR TITLE
Add AppendAttribute, Add IsWriteEnabled check for GetAttributes

### PR DESCRIPTION
**Pull request by lavantgarde**

### CODE REVIEW

File: AutoCADCommands/Test.cs [hunk 1]
```diff
@@ -1003,12 +1003,20 @@ public void TestPolygonMesh()
         }
 
         [CommandMethod("TestAddAttribute")]
-        public void TestAddAttribute()
+        public void TestAppendAttribute()
         {
             var iId = Draw.Insert("test", Point3d.Origin);
```
Comment (luanshixia) [inline AutoCADCommands/Test.cs]: Let's not prefix variable names. `iId` -> `id` or `insertId`

File: AutoCADCommands/DbHelper.cs [hunk 2]
```diff
@@ -567,6 +567,22 @@ internal static string GetCode(ObjectId id, Database db)
...
         #region block attribute
 
+        /// <summary>
+        /// Returns a dictionary of a block reference's block attribute names and ObjectIds.
+        /// </summary>
+        /// <param name="blockReference">The block reference.</param>
+        /// <returns></returns>
+        public static Dictionary<string, ObjectId> GetBlockAttributeIds(this BlockReference blockReference)
+        {
+            Dictionary<string, ObjectId> attrs = new Dictionary<string, ObjectId>();
```
Comment (luanshixia) [inline AutoCADCommands/DbHelper.cs]: Let's use `var`.

**Review by luanshixia — COMMENTED** [review]
Thanks!

File: AutoCADCommands/Test.cs (hunk 1 above)
Comment (lavantgarde) [reply AutoCADCommands/Test.cs]: oops, good catch.

**Pull request merged**